spec/support/shared_contexts/with_contact_tag.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+RSpec.shared_context 'with contact tag params', with_contact_tag_params: true do
+  include_context 'with existing contact'
+  include_context 'with existing tag'
+
+  let!(:contact_id) { contact[:id] }
+  let!(:tag_id)     { tag[:id] }
+  let(:contact_tag_params) do
+    {
+      contact: contact_id,
+      tag: tag_id
+    }
+  end
+
+  let(:expected_contact_tag_response) do
+    contact_tag_params
+  end
+end
+
+RSpec.shared_context 'with existing contact tag', with_existing_contact_tag: true do
+  include_context 'with contact tag params'
+
+  let!(:contact_tag) do
+    response = client.create_contact_tag(contact_tag_params)
+    response.fetch(:contact_tag) { raise "HELL (contact tag creation failed) #{response}" }
```

Comment:
As you can see the notation is translated back into underscore or snake_case on the way back.